In [ ]:
# homework1-MalloryHelfenbein
# homework1-MalloryHelfenbein
# homework1-MalloryHelfenbein


# Project 0- completed

#Problem 1- completed once other problems are completed 



# Problem 2 part a and b

In [21]:
n=10030508582493
import math

def PrimeFactor(n): 
    #So this shows the number of times that 2 is a prime factor of the number "n" and prints them out
    while n % 2 == 0:
        print (2),
        n = n/2
        # Once the while loop is ended, n must be odd at this point so move on to the next division
   
for i in range(3, int(math.sqrt(n))+1,2):
        
        while n%i==0:
            print (i),
            n=n/i
    #this will print out every number(i) that goes into number (n) until it is no longer evenly divisible. 


3
7
7789
7789
7873


b) This shows that the largest prime factor of 10,030,508,582,493 is 7873

# Problem 3

In [26]:
def factorial (x):
    if x == 0: 
        ans = 1
    else: 
        ans = 1
        for i in range(1,x+1):
            ans = ans * i
    return ans

In [27]:
factorial (5)

120

# Problem 4

In [31]:
num= #this is where you enter the number
reversenum=0
while(num>0):
    dig=num%10
    reversenum=reversenum*10+dig
    num=num//10
if (num==reversenum):
    print ("Palindrome")
else:
    print ("Not Palindrome")
#I got some of my lines of code from a website called mastering python
# Also I am not sure if I did this part right, I was a little confused

Not Palindrome


part A) I am not sure about this one

part B) 906609, 913 * 993